lib/tests/misc.nix

```diff
@@ -312,6 +312,34 @@ runTests {
     expected = true;
   };
 
+  testToInt = testAllTrue [
```

Comment:
Perhaps add two tests which `toInt` fails to parse. For example, an empty string and a number with spaces between
- ""
- "123  123"
- " 0 123 "

Reply:
This should be possible via tryEval. It's done for `toInt` later to demonstrate the functionality of tryEval itself. 

Example: https://github.com/NixOS/nixpkgs/blob/79196b6ab885df354dc8c31f9dbbf1ed28c76f17/lib/tests/misc.nix#L405

I'll try and make this change tonight.

Reply:
I think I've found another bug while working on this. It looks like some errors escape `builtins.tryEval` while others do not. I'll be kicking out a new issue to deal with this and I'll probably be blocking this PR until that issue is resolved.

Reply:
ok actually apparently this is expected behavior according to NixOS/nix#5564. I'm not sure how to implement these tests otherwise. Maybe @edolstra has some insight on how to best move forward?

Reply:
The failing test is commented out but if uncommented, it errors escaping the tryEval.

https://github.com/NixOS/nixpkgs/pull/173949/files#diff-9c1b3fea9dfcb1a6c1154253a234e794b1c2e49c4c84421bac388786ad0615deR343-R346

Reply:
https://github.com/NixOS/nixpkgs/pull/173949#discussion_r879121324

Reply:
I think I've found a solution. As is I think toInt should now always get caught by the throw from the regex rather than ever making it into the parser.